### PR TITLE
[site/libraries] Refactored some for-loops to foreach... 

### PR DESCRIPTION
**Pull request by frankmayer**
As the title says...

### CODE REVIEW

File: libraries/cms/html/access.php [hunk 1]
```diff
@@ -132,9 +132,8 @@ public static function usergroups($name, $selected, $checkSuperAdmin = false)
 
 		$html = array();
 
-		for ($i = 0, $n = count($groups); $i < $n; $i++)
+		foreach ($groups as &$item)
```
Comment (andrepereiradasilva) [inline libraries/cms/html/access.php]: i don't think you need to pass as reference here

File: libraries/cms/html/access.php [hunk 2]
```diff
@@ -192,25 +191,26 @@ public static function actions($name, $selected, $component, $section = 'global'
...
-
-			// Build the HTML for the item.
-			$html[] = '	<li>';
-			$html[] = '		<input type="checkbox" name="' . $name . '[]" value="' . $item->id . '" id="' . $eid . '"';
-			$html[] = '			' . $checked . ' />';
-			$html[] = '		<label for="' . $eid . '">';
-			$html[] = '			' . JText::_($item->title);
-			$html[] = '		</label>';
-			$html[] = '	</li>';
+			foreach ($actions as &$item)
```
Comment (andrepereiradasilva) [inline libraries/cms/html/access.php]: i don't think you need to pass as reference here

File: libraries/cms/html/access.php [hunk 3]
```diff
@@ -192,25 +191,26 @@ public static function actions($name, $selected, $component, $section = 'global'
 			"/access/section[@name='" . $section . "']/"
 		);
 
-		$html = array();
+		$html   = array();
 		$html[] = '<ul class="checklist access-actions">';
 
-		for ($i = 0, $n = count($actions); $i < $n; $i++)
+		if ((bool) $actions)
```
Comment (andrepereiradasilva) [inline libraries/cms/html/access.php]: why add this?

File: libraries/cms/html/access.php (hunk 1 above)
Comment (frankmayer) [inline libraries/cms/html/access.php]: Right, thanks!

File: libraries/cms/html/access.php (hunk 3 above)
Comment (frankmayer) [inline libraries/cms/html/access.php]: I was under the impression that foreach might throw a warning or notice, if fed with false, but I cannot find any evidence of this now. Removing...

File: libraries/cms/html/access.php (hunk 2 above)
Comment (frankmayer) [inline libraries/cms/html/access.php]: Right, thanks!

**Comment by rdeutz** [comment]
see my comment #12299

**Pull request closed without merging**